sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -529,6 +530,11 @@ object LikeSimplification extends Rule[LogicalPlan] {
           StartsWith(l, Literal(pattern))
         case endsWith(pattern) =>
           EndsWith(l, Literal(pattern))
+        // 'a%a' pattern is basically same with 'a%' && '%a'.
+        // However, the additional `Length` condition is required to prevent 'a' match 'a%a'.
+        case startsAndEndsWith(prefix, postfix) if !prefix.endsWith("\\") =>
```

Comment:
while you are at it, can you rename "pattern" in the startsWith case to prefix, and endsWith to suffix?